autometa/taxonomy/database.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env python
+
+import logging
+import string
+
+from abc import ABC, abstractmethod
+from typing import Dict, Set, Tuple, List, Union, Iterable
+from autometa.common.exceptions import DatabaseOutOfSyncError
+
+import pandas as pd
+
+
+logger = logging.getLogger(__name__)
+
+
+class TaxonomyDatabase(ABC):
```

Comment:
@WiscEvan If I remember correctly this function/class was written by you and I don't think I fully understand its role. It will be great if you can add the docstring instead 😄